test/js/web/timers/setImmediate.test.js

```diff
@@ -35,7 +35,6 @@ it("clearImmediate", async () => {
   var called = false;
   const id = setImmediate(() => {
     called = true;
-    expect.unreachable();
```

Comment:
The expect would fail outside the test. Intent was to change it so that the failure is reported inside the test. I changed it to use `Promise.withResolvers` instead